commercetools/resource_custom_application.go

```diff
@@ -0,0 +1,422 @@
+package commercetools
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/machinebox/graphql"
+)
+
+func resourceCustomApplication() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCustomApplicationCreate,
+		Read:   resourceCustomApplicationRead,
+		Update: resourceCustomApplicationUpdate,
+		Delete: resourceCustomApplicationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"is_active": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			// NOTE: currently the terraform SDK does not yet support nested object maps,
+			// even though terraform 0.12 has support for that. As a workaround we can use
+			// a `TypeList` with a `MaxItems` of 1
+			// https://github.com/hashicorp/terraform-plugin-sdk/issues/62
+			// https://github.com/hashicorp/terraform-plugin-sdk/issues/155.
+			"navbar_menu": {
+				Type:     schema.TypeList,
```

Comment:
In JSON, this is an object, not a list. However, as far as I understand, nested object maps are not supported yet by the terraform sdk. See comment above